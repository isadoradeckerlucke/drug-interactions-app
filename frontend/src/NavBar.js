import React from "react";
import { useDispatch, useSelector } from "react-redux";
import Navbar from "react-bootstrap/Navbar";
import Nav from "react-bootstrap/Nav";
import Button from "react-bootstrap/Button";
import { logOut } from "./actions/logout";

function NavBar() {
  const currentUser = useSelector((st) => st.currentUser);
  const dispatch = useDispatch();

  function logOutUser(e) {
    dispatch(logOut());
    window.location.reload();
  }

  return (
    <Navbar bg="dark" expand="lg" variant="dark">
      <Navbar.Brand href="/">Drug Interaction Hub</Navbar.Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse id="basic-navbar-nav">
        <Nav className="mr-auto">
          <Nav.Link href="/interactions">Search Drug Interactions</Nav.Link>
          {currentUser.isLoggedIn ? (
            <Nav>
              <Nav.Link href={"/users/" + currentUser.currentUser.username}>
                My Saved Drugs
              </Nav.Link>
              <Button onClick={logOutUser} variant="outline-danger">
                Log Out
              </Button>
            </Nav>
          ) : (
            <Nav>
              <Nav.Link href="/signup">Sign Up</Nav.Link>
              <Nav.Link href="/login">Log In</Nav.Link>
            </Nav>
          )}
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
}

export default NavBar;
