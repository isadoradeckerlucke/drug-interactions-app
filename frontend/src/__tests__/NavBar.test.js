import React from "react";
import { render } from "@testing-library/react";
import NavBar from "../NavBar";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import configureStore from "redux-mock-store";

describe("nav bar works as expected", function () {
  const initialState = {
    searchTerm: { searchTerm: "adderall" },
    currentUser: {
      isLoggedIn: true,
      currentUser: {
        username: "testuser",
        email: "testuser@gmail.com",
        savedDrugs: [],
      },
    },
  };
  const mockStore = configureStore();
  let store;

  it("renders without crashing", function () {
    store = mockStore(initialState);
    render(
      <MemoryRouter>
        <Provider store={store}>
          <NavBar />
        </Provider>
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    store = mockStore(initialState);
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <NavBar />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
