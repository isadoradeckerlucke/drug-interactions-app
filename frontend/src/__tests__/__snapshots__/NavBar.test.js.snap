// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav bar works as expected matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Drug Interaction Hub
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        class="mr-auto navbar-nav"
      >
        <a
          class="nav-link"
          data-rb-event-key="/interactions"
          href="/interactions"
        >
          Search Drug Interactions
        </a>
        <div
          class="navbar-nav"
        >
          <a
            class="nav-link"
            data-rb-event-key="/users/testuser"
            href="/users/testuser"
          >
            My Saved Drugs
          </a>
          <button
            class="btn btn-outline-danger"
            type="button"
          >
            Log Out
          </button>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
